strawberryfields/circuitspecs/chip0.py

```diff
@@ -51,7 +56,42 @@ class Chip0Specs(CircuitSpecs):
         Rgate({internal_phase_0}) | [2]
         BSgate(pi/4, pi/2) | [2, 3]
 
+        # final local phases
+        Rgate({phi0}) | 0
+        Rgate({phi0}) | 1
+        Rgate({phi0}) | 2
+        Rgate({phi0}) | 3
+
         # Measurement in Fock basis
         MeasureFock() | [0, 1, 2, 3]
         """
     )
+
+    def compile(self, seq):
+        # first do general GBS compilation to make sure
+        # Fock measurements are correct
+        seq = GBSSpecs().compile(seq)
```

Comment:
Another option: inherit GBSSpec, call `super().compile()` here?

Reply:
This makes it difficult to call `Circuit Spec.compile` for the final topology check, though

Reply:
@josh146 how come?

Reply:
Because Python doesn't really provide a way to call methods from the grandfather class. The recommended approach is to invoke the grandfather class directly, but this doesn't work in this case, since `CircuitSpec` is an abstract base class and cannot be instantiated.

Reply:
I think you can call a non-abstract method of an ABC, `ABC.methodname(self_param, other_params)`

Reply:
I tried this, I got `TypeError: Can't instantiate abstract class CircuitSpecs with abstract methods`